src/pages/resultsView/plots/PlotsTabUtils.tsx

```diff
@@ -132,22 +130,47 @@ export interface IScatterPlotSampleData {
     copyNumberAlterations: AnnotatedNumericGeneMolecularData[];
 }
 
-export interface IScatterPlotData extends IScatterPlotSampleData, IBaseScatterPlotData {};
+export interface IThresholdDataPoint {
+    thresholdType?:'>'|'<'|undefined;
+}
+export interface IThresholdsDataPoint {
+    xThresholdType?: '>'|'<'|undefined;
+    yThresholdType?: '>'|'<'|undefined;
+}
 
-export interface IBoxScatterPlotPoint extends IScatterPlotSampleData {
+export interface IBoxScatterPlotPoint extends IPlotSampleData, IThresholdDataPoint {
     category:string;
     value:number;
     jitter:number;
 }
 
+export interface IScatterPlotData extends IPlotSampleData, IBaseScatterPlotData, IThresholdsDataPoint {};
+export interface IWaterfallPlotData extends IPlotSampleData, IBaseWaterfallPlotData, IThresholdDataPoint {};
+
+export interface IAxisLogScaleParams {
+    label:string;
+
+    // Function for transforming `number` into its log-scale representation
+    // `offset` is a constant added to the value before transformation (used
+    // for transformation of data sets with negative numbers)
+    fLogScale:((x:number, offset:number)=>number);
+
+    // Function for back-transforming `number` transformed with fLogScale
+    // function into its linear-scale representation
+    fInvLogScale:((x:number)=>number);
```

Comment:
just occurred to me, dont you need offset to be a part of this too

Reply:
Added offset parameters as optional parameters to the interfaces. For the log10-backtransformations this parameter is now also used. Unit tests were added.